lib/ruby_lsp/listeners/document_highlight.rb

```diff
@@ -252,13 +273,34 @@ def on_required_parameter_node_enter(node)
 
       sig { params(node: Prism::ClassNode).void }
       def on_class_node_enter(node)
+        if @target.is_a?(Prism::ClassNode)
+          class_keyword_loc = node.class_keyword_loc
+          end_keyword_loc = node_closing_keyword_loc(node)
+          if end_keyword_loc && (covers_target_position?(class_keyword_loc) || covers_target_position?(end_keyword_loc))
+            add_highlight(Constant::DocumentHighlightKind::TEXT, class_keyword_loc)
+            add_highlight(Constant::DocumentHighlightKind::TEXT, end_keyword_loc)
```

Comment:
This logic seems to be repeated for all nodes. Can we extract this into a method that accepts both locations and then add the highlights if necessary?